Add quotes around nexthop and dst-ip fields

tssurya opened the pull request:
**- What this PR does and why is it needed**

While providing ipv6 addresses in `nexthop` and `dst-ip` fields into the
nbctl calls, it fails with the error `unexpected ":" parsing string`
during hbo pod deletion. This leads to stale bfd entries that don't
get cleaned up.

Signed-off-by: Surya Seetharaman <suryaseetharaman.9@gmail.com>


**- How to verify it**
Without the PR:

```
I0820 10:59:13.626354      48 pods.go:94] Deleting pod: t1/client-on-ovn-worker
I0820 10:59:13.626393      48 address_set.go:481] (d79acab8-7d39-47cc-8682-ab7bf0d80bb4/t1_v6/a3345265888189311907) deleting IP "fd00:10:244:1::7" from address set
I0820 10:59:13.626492      48 ovs.go:207] exec(188): /usr/bin/ovn-nbctl --timeout=15 remove address_set d79acab8-7d39-47cc-8682-ab7bf0d80bb4 addresses "fd00:10:244:1::7"
I0820 10:59:13.630174      48 ovs.go:210] exec(188): stdout: ""
I0820 10:59:13.630206      48 ovs.go:211] exec(188): stderr: ""
I0820 10:59:13.631453      48 ovs.go:207] exec(189): /usr/bin/ovn-nbctl --timeout=15 --if-exists --policy=src-ip lr-route-del GR_ovn-control-plane fd00:10:244:1::7/128 fd2e:6f44:5dd8::89
I0820 10:59:13.635284      48 ovs.go:210] exec(189): stdout: ""
I0820 10:59:13.635316      48 ovs.go:211] exec(189): stderr: ""
I0820 10:59:13.635384      48 ovs.go:207] exec(190): /usr/bin/ovn-nbctl --timeout=15 --format=csv --data=bare --no-heading --columns=bfd find Logical_Router_Static_Route output_port=rtoe-GR_ovn-control-plane nexthop=fd2e:6f44:5dd8::89 bfd!=[]
I0820 10:59:13.638725      48 ovs.go:210] exec(190): stdout: ""
I0820 10:59:13.638761      48 ovs.go:211] exec(190): stderr: "ovn-nbctl: fd2e:6f44:5dd8::89: unexpected \":\" parsing string\n"
I0820 10:59:13.638773      48 ovs.go:213] exec(190): err: exit status 1
E0820 10:59:13.638801      48 egressgw.go:540] cleanUpBFDEntry: failed to list routes for rtoe-GR_ovn-control-plane, stderr: "fd2e:6f44:5dd8::89", (OVN command '/usr/bin/ovn-nbctl --timeout=15 --format=csv --data=bare --no-heading --columns=bfd find Logical_Router_Static_Route output_port=rtoe-GR_ovn-control-plane nexthop=fd2e:6f44:5dd8::89 bfd!=[]' failed: exit status 1)%!(EXTRA string=ovn-nbctl: fd2e:6f44:5dd8::89: unexpected ":" parsing string
)
```
and

```
I0820 11:14:13.773141      50 pods.go:92] Deleting pod: t1/client-on-ovn-worker
I0820 11:14:13.773179      50 address_set.go:523] (e1da2d4e-0a64-473d-91a1-4849694f19f3/t1_v6/a3345265888189311907) deleting IP [fd00:10:244:1::7] from address set
I0820 11:14:13.774855      50 ovs.go:209] exec(86): /usr/bin/ovn-nbctl --timeout=15 --if-exists --policy=src-ip lr-route-del GR_ovn-control-plane fd00:10:244:1::7/128 fd2e:6f44:5dd8::89
I0820 11:14:13.780303      50 ovs.go:212] exec(86): stdout: ""
I0820 11:14:13.781026      50 ovs.go:213] exec(86): stderr: ""
I0820 11:14:13.781179      50 ovs.go:209] exec(87): /usr/bin/ovn-nbctl --timeout=15 --format=csv --data=bare --no-heading --columns=bfd find Logical_Router_Static_Route output_port=rtoe-GR_ovn-control-plane nexthop="fd2e:6f44:5dd8::89" bfd!=[]
I0820 11:14:13.786979      50 ovs.go:212] exec(87): stdout: ""
I0820 11:14:13.787009      50 ovs.go:213] exec(87): stderr: ""
I0820 11:14:13.787069      50 ovs.go:209] exec(88): /usr/bin/ovn-nbctl --timeout=15 --format=csv --data=bare --no-heading --columns=_uuid find BFD logical_port=rtoe-GR_ovn-control-plane dst_ip=fd2e:6f44:5dd8::89
I0820 11:14:13.790103      50 ovs.go:212] exec(88): stdout: ""
I0820 11:14:13.790190      50 ovs.go:213] exec(88): stderr: "ovn-nbctl: fd2e:6f44:5dd8::89: unexpected \":\" parsing string\n"
I0820 11:14:13.790222      50 ovs.go:215] exec(88): err: exit status 1
E0820 11:14:13.790336      50 egressgw.go:537] Failed to list routes for GR_ovn-control-plane, stderr: "OVN command '/usr/bin/ovn-nbctl --timeout=15 --format=csv --data=bare --no-heading --columns=_uuid find BFD logical_port=rtoe-GR_ovn-control-plane dst_ip=fd2e:6f44:5dd8::89' failed: exit status 1", (ovn-nbctl: fd2e:6f44:5dd8::89: unexpected ":" parsing string
)
```

With the PR:
```
I0820 11:39:25.337712      52 pods.go:92] Deleting pod: t1/client-on-ovn-worker
I0820 11:39:25.337807      52 address_set.go:523] (07240eec-45c1-487a-9708-4a9bf72d6739/t1_v6/a3345265888189311907) deleting IP [fd00:10:244:2::8] from address set
I0820 11:39:25.341663      52 ovs.go:209] exec(97): /usr/bin/ovn-nbctl --timeout=15 --if-exists --policy=src-ip lr-route-del GR_ovn-control-plane fd00:10:244:2::8/128 fd2e:6f44:5dd8::89
I0820 11:39:25.345647      52 ovs.go:212] exec(97): stdout: ""
I0820 11:39:25.346212      52 ovs.go:213] exec(97): stderr: ""
I0820 11:39:25.346359      52 ovs.go:209] exec(98): /usr/bin/ovn-nbctl --timeout=15 --format=csv --data=bare --no-heading --columns=bfd find Logical_Router_Static_Route output_port=rtoe-GR_ovn-control-plane nexthop="fd2e:6f44:5dd8::89" bfd!=[]
I0820 11:39:25.349843      52 ovs.go:212] exec(98): stdout: ""
I0820 11:39:25.349896      52 ovs.go:213] exec(98): stderr: ""
I0820 11:39:25.350032      52 ovs.go:209] exec(99): /usr/bin/ovn-nbctl --timeout=15 --format=csv --data=bare --no-heading --columns=_uuid find BFD logical_port=rtoe-GR_ovn-control-plane dst_ip="fd2e:6f44:5dd8::89"
I0820 11:39:25.353364      52 ovs.go:212] exec(99): stdout: "92f421a6-30f7-49aa-914c-3056a30bed47\n"
I0820 11:39:25.353401      52 ovs.go:213] exec(99): stderr: ""
I0820 11:39:25.353464      52 ovs.go:209] exec(100): /usr/bin/ovn-nbctl --timeout=15 --if-exists destroy BFD 92f421a6-30f7-49aa-914c-3056a30bed47
I0820 11:39:25.356819      52 ovs.go:212] exec(100): stdout: ""
I0820 11:39:25.356844      52 ovs.go:213] exec(100): stderr: ""
I0820 11:39:25.356936      52 ovs.go:209] exec(101): /usr/bin/ovn-nbctl --timeout=15 --if-exists --policy=src-ip lr-route-del GR_ovn-control-plane fd00:10:244:2::8/128 fd2e:6f44:5dd8::76
I0820 11:39:25.360766      52 ovs.go:212] exec(101): stdout: ""
I0820 11:39:25.361056      52 ovs.go:213] exec(101): stderr: ""
I0820 11:39:25.361374      52 ovs.go:209] exec(102): /usr/bin/ovn-nbctl --timeout=15 --format=csv --data=bare --no-heading --columns=bfd find Logical_Router_Static_Route output_port=rtoe-GR_ovn-control-plane nexthop="fd2e:6f44:5dd8::76" bfd!=[]
I0820 11:39:25.366050      52 ovs.go:212] exec(102): stdout: ""
I0820 11:39:25.366081      52 ovs.go:213] exec(102): stderr: ""
I0820 11:39:25.366163      52 ovs.go:209] exec(103): /usr/bin/ovn-nbctl --timeout=15 --format=csv --data=bare --no-heading --columns=_uuid find BFD logical_port=rtoe-GR_ovn-control-plane dst_ip="fd2e:6f44:5dd8::76"
I0820 11:39:25.369571      52 ovs.go:212] exec(103): stdout: "46fdae8c-abe3-4853-b48a-f972be776744\n"
I0820 11:39:25.369603      52 ovs.go:213] exec(103): stderr: ""
I0820 11:39:25.369657      52 ovs.go:209] exec(104): /usr/bin/ovn-nbctl --timeout=15 --if-exists destroy BFD 46fdae8c-abe3-4853-b48a-f972be776744
I0820 11:39:25.373585      52 ovs.go:212] exec(104): stdout: ""
I0820 11:39:25.373616      52 ovs.go:213] exec(104): stderr: ""
```

**- Description for the changelog**
`Fixes bfd entry deletion in ipv6 mode`

dcbw commented:
@tssurya looks good, but need to update the testcases to match:

```
2021-08-20T11:59:55.7193961Z W0820 11:59:55.718917   17987 exec.go:265] Unexpected command: /fake-bin/ovn-nbctl --timeout=15 --format=csv --data=bare --no-heading --columns=bfd find Logical_Router_Static_Route output_port=rtoe-GR_node1 nexthop="9.0.0.1" bfd!=[]
2021-08-20T11:59:55.7195201Z 
2021-08-20T11:59:55.7195769Z Executed commands do not match expected commands!
2021-08-20T11:59:55.7197313Z Executed unexpected command /fake-bin/ovn-nbctl --timeout=15 --format=csv --data=bare --no-heading --columns=bfd find Logical_Router_Static_Route output_port=rtoe-GR_node1 nexthop="9.0.0.1" bfd!=[]
2021-08-20T11:59:55.7198014Z 
2021-08-20T11:59:55.7199072Z [00]   /fake-bin/ovn-nbctl --timeout=15 --may-exist --policy=src-ip --ecmp-symmetric-reply lr-route-add GR_node1 10.128.1.3/32 9.0.0.1 rtoe-GR_node1
2021-08-20T11:59:55.7200434Z [01]   /fake-bin/ovn-nbctl --timeout=15 --if-exists --policy=src-ip lr-route-del GR_node1 10.128.1.3/32 9.0.0.1
2021-08-20T11:59:55.7201874Z [02] ! /fake-bin/ovn-nbctl --timeout=15 --format=csv --data=bare --no-heading --columns=bfd find Logical_Router_Static_Route output_port=rtoe-GR_node1 nexthop=9.0.0.1 bfd!=[]
2021-08-20T11:59:55.7203762Z [03] ! /fake-bin/ovn-nbctl --timeout=15 --format=csv --data=bare --no-heading --columns=_uuid find BFD logical_port=rtoe-GR_node1 dst_ip=9.0.0.1
2021-08-20T11:59:55.7204437Z goroutine 13773 [running]:
```

tssurya commented:
> @tssurya looks good, but need to update the testcases to match:
> 
> ```
> 2021-08-20T11:59:55.7193961Z W0820 11:59:55.718917   17987 exec.go:265] Unexpected command: /fake-bin/ovn-nbctl --timeout=15 --format=csv --data=bare --no-heading --columns=bfd find Logical_Router_Static_Route output_port=rtoe-GR_node1 nexthop="9.0.0.1" bfd!=[]
> 2021-08-20T11:59:55.7195201Z 
> 2021-08-20T11:59:55.7195769Z Executed commands do not match expected commands!
> 2021-08-20T11:59:55.7197313Z Executed unexpected command /fake-bin/ovn-nbctl --timeout=15 --format=csv --data=bare --no-heading --columns=bfd find Logical_Router_Static_Route output_port=rtoe-GR_node1 nexthop="9.0.0.1" bfd!=[]
> 2021-08-20T11:59:55.7198014Z 
> 2021-08-20T11:59:55.7199072Z [00]   /fake-bin/ovn-nbctl --timeout=15 --may-exist --policy=src-ip --ecmp-symmetric-reply lr-route-add GR_node1 10.128.1.3/32 9.0.0.1 rtoe-GR_node1
> 2021-08-20T11:59:55.7200434Z [01]   /fake-bin/ovn-nbctl --timeout=15 --if-exists --policy=src-ip lr-route-del GR_node1 10.128.1.3/32 9.0.0.1
> 2021-08-20T11:59:55.7201874Z [02] ! /fake-bin/ovn-nbctl --timeout=15 --format=csv --data=bare --no-heading --columns=bfd find Logical_Router_Static_Route output_port=rtoe-GR_node1 nexthop=9.0.0.1 bfd!=[]
> 2021-08-20T11:59:55.7203762Z [03] ! /fake-bin/ovn-nbctl --timeout=15 --format=csv --data=bare --no-heading --columns=_uuid find BFD logical_port=rtoe-GR_node1 dst_ip=9.0.0.1
> 2021-08-20T11:59:55.7204437Z goroutine 13773 [running]:
> ```

yea my bad, updated the PR and added an IPV6 test as well to see how it would look like. Probably these things would get better with the libovsdb stuff but we'd need this backported to 4.8 at least.

tssurya commented:
/retest

tssurya commented:
unrelated failure for the CI (but that control-place, noHA, snatGW, local, 1br) seems to be failing a lot! Need to look into that.
This specific PR can go in.
```
Latency metrics for node ovn-worker2
Aug 20 21:54:22.548: INFO: Waiting up to 3m0s for all (but 0) nodes to be ready
�[1mSTEP�[0m: Destroying namespace "netflow-test-1223" for this suite.
[AfterEach] e2e br-int NetFlow export validation
  /home/runner/work/ovn-kubernetes/ovn-kubernetes/test/e2e/e2e.go:2451
       - /home/runner/work/ovn-kubernetes/ovn-kubernetes/test/e2e/e2e.go:2460
Aug 20 21:59:23.781: ovnkube-node pods are not ready: timed out waiting for the condition
/home/runner/go/pkg/mod/github.com/onsi/ginkgo@v1.12.0/internal/leafnodes/runner.go:113
       - Failure
       - [BeforeEach] e2e br-int NetFlow export validation
  /home/runner/go/pkg/mod/k8s.io/kubernetes@v1.20.2/test/e2e/framework/framework.go:174
�[1mSTEP�[0m: Creating a kubernetes client
Aug 20 21:54:22.946: INFO: >>> kubeConfig: /home/runner/admin.conf
�[1mSTEP�[0m: Building a namespace api object, basename netflow-test
�[1mSTEP�[0m: Waiting for a default service account to be provisioned in namespace
[It] Should validate NetFlow data of br-int is sent to an external gateway
  /home/runner/work/ovn-kubernetes/ovn-kubernetes/test/e2e/e2e.go:2460
Aug 20 21:54:23.675: INFO: Setting OVN_NETFLOW_TARGETS environment variable value to NetFlow collector IP 172.18.0.5
Aug 20 21:54:23.675: INFO: Running '/usr/local/bin/kubectl --server=https://10.1.1.171:11337 --kubeconfig=/home/runner/admin.conf --namespace=ovn-kubernetes set env daemonset/ovnkube-node -c ovnkube-node OVN_NETFLOW_TARGETS=172.18.0.5:2056'
Aug 20 21:54:23.770: INFO: stderr: ""
Aug 20 21:54:23.770: INFO: stdout: ""
Aug 20 21:59:23.781: FAIL: ovnkube-node pods are not ready: timed out waiting for the condition
```
